app/components/contacts/utils.ts

```diff
@@ -17,35 +17,41 @@ export const validate = (
   oldAddress?: string
 ): ValidateError | null => {
   let filteredContacts = contacts;
+  const trimmedAddress = address.trim();
 
   if (oldAddress) {
     filteredContacts = contacts.filter(
       (contact) => contact.address !== oldAddress
     );
   }
+  const trimmedNickname = nickname.trim();
 
   const nicknameRegex = /^[a-zA-Z0-9_-]*$/;
-  if (!nicknameRegex.test(nickname) || !nickname) {
+  if (!nicknameRegex.test(trimmedNickname) || !trimmedNickname) {
     return {
       type: 'name',
-      message: `Nickname: ${nickname} is missing or invalid`,
+      message: `Nickname: ${trimmedNickname} is missing or invalid`,
```

Comment:
Actually, there's no info anywhere about what is valid and what is not. I suggest adding such info in a tooltip or placeholder text in the nickname field. 
It's shooting in the dark right now and hoping smapp will not refuse my contact name.